### PR TITLE
CATALOG.md: build command library (fix #1166)

### LINKED ISSUE #1166: [Road to nativefier-catalog] First have a "Build Commands Library" md file compiling "known minimal good configurations" per site
I was wondering if we could enable the Wiki in Github or maybe just use a "Feature Request type Issue" here where we can post the commands we use as a reference for everyone. For example below is what I do for Outlook Web and for Udemy:

#### Command String
```
nativefier https://outlook.office.com/mail 
--name OutlookWA 
--internal-urls ".*?(outlook.live.com|outlook.office365.com|outlook.office.com|login.microsoftonline.com).*?" 
--file-download-options "{\"saveAs\": true}" 
--disable-dev-tools 
--disk-cache-size 0 
--browserwindow-options "{ \"webPreferences\": { \"webviewTag\": true, \"nodeIntegration\": true, \"nodeIntegrationInSubFrames\": true, \"nativeWindowOpen\": true } }"
```
#### Notes:
In order to have the "pop out" window functionality when creating an email, the `--browserwindow-options` is needed.

#### Command String
```
nativefier https://www.udemy.com/ 
--name Udemy 
--internal-urls ".*?udemy.*?" 
--file-download-options "{\"saveAs\": true}" 
--disable-dev-tools 
--disk-cache-size 0 
--icon udemy.ico 
--widevine
```
#### Notes:
Most videos will work, but to be sure everything works you are better off using the `--widevine` version and sign the app afterwards. See this post: https://github.com/nativefier/nativefier/issues/1147#issuecomment-828750362

### PR DESCRIPTION
**Pull request by deciacco**
Added the CATALOG.md file and updated the README.md file Usage section to link to the new file.

### CODE REVIEW

File: CATALOG.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+* * *
+
+## Outlook Web
+
+### Windows Command
+```
+nativefier https://outlook.office.com/mail 
+--name OutlookWA 
+--internal-urls ".*?(outlook.live.com|outlook.office365.com|outlook.office.com|login.microsoftonline.com).*?"
```
Comment (TheCleric) [inline CATALOG.md:20]: I'd say that `login.microsoftonline.com` should be removed here, and probably just added to the internal login page list, but that could probably be covered in a separate PR.

File: CATALOG.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+- ... but let's not have all the other flags that are a matter of personal preference (e.g. `--disable-dev-tools` or `--disk-cache-size`).
+
+* * *
+
+## Outlook Web
+
+### Windows Command
+```
+nativefier https://outlook.office.com/mail 
+--name OutlookWA
```
Comment (TheCleric) [inline CATALOG.md:19]: I think for these I'd leave `--name` off and either let it be inferred, or allow the user to decide, since this won't change the app behavior in any way.

File: CATALOG.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+### Notes
+
+`--browserwindow-options` -- This is needed in order to allow the window to pop out when creating/editing an email.
+
+## Udemy
+
+### Windows Command
+```
+nativefier https://www.udemy.com/ 
+--name Udemy
```
Comment (TheCleric) [inline CATALOG.md:34]: Same as above regarding `name`

File: README.md [hunk 4]
```diff
@@ -57,6 +57,10 @@ to learn about other command-line flags usable to configure the packaged app.
 To have high-resolution icons used by default for an app/domain, please
 contribute to the [icon repository](https://github.com/nativefier/nativefier-icons)!
 
+### Build Commands Library
+
+For a list of build commands contributed by the nativefier community take a look at the [CATALOG.md file](CATALOG.md)
```
Comment (TheCleric) [inline README.md:62]: This is pretty cool! I didn't realize that you could direct link to other .md files like this. We should probably fix all our other links to be like this. 😆

**Review by TheCleric — CHANGES_REQUESTED** [review]

**Comment by TheCleric** [comment]
Looks pretty good! Just a few small questions/comments.

File: CATALOG.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
+# Build Commands Library
```
Comment (ronjouch) [inline CATALOG.md:1]: Let's keep consistent naming between filename and title. "Catalog" everywhere or "Library" everywhere or "Example" everywhere. All are fine by me, your choice

File: CATALOG.md (hunk 2 above)
Comment (deciacco) [reply CATALOG.md:19]: Makes sense... I will remove!!

File: CATALOG.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+Below you'll find a list of build commands contributed from the nativefier community.
```
Comment (ronjouch) [inline CATALOG.md:3]:
```suggestion
Below you'll find a list of build commands contributed by the Nativefier community. They are here as examples, and to help you nativefy "complicated" apps that ask a bit of elbow grease to work.
```

File: CATALOG.md (hunk 1 above)
Comment (deciacco) [reply CATALOG.md:20]: I will remove... not sure I know what the internal login page list is tho...

File: CATALOG.md (hunk 3 above)
Comment (deciacco) [reply CATALOG.md:34]: will remove here as well...

File: README.md (hunk 4 above)
Comment (deciacco) [reply README.md:62]: to be honest I didn't either. I just saw that it was done in the readme file...

File: CATALOG.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+If you'd like to add to this catalog, please keep the following in mind:
+
+Library commands should be the strict necessary to make an app work. So, for example,
```
Comment (ronjouch) [inline CATALOG.md:7]: Compacting these two things into a single point, and adding a point asking to not add trivial sites.

```suggestion
- Only add sites that require something special! No need to document here that `simplesite.com` works with a simple `nativefier simplesite.com` 🙂
- If you'd like to add to this catalog, please add commands with the *strict necessary* to make an app work. So, for example,
```

File: CATALOG.md (hunk 1 above)
Comment (TheCleric) [reply CATALOG.md:20]: It's in app/src/helpers/helpers.ts (and there are tests in app/src/helpers/helpers.test.ts

Again, we can handle this in a separate PR.

File: CATALOG.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+- Yes to have `--widevine` in Udemy. Yes to `--internal-urls` and `--browserwindow-options` for Outlook Web. Yes to a name and icon for both...
+- ... but let's not have all the other flags that are a matter of personal preference (e.g. `--disable-dev-tools` or `--disk-cache-size`).
+
+* * *
+
+## Outlook Web
+
+### Windows Command
```
Comment (ronjouch) [inline CATALOG.md:16]: I wouldn't keep these headings, they add repeated noise, and consume vertical space. Just the example in triple backticks and a list of notes is enough.

File: README.md [hunk 9]
```diff
@@ -57,6 +57,10 @@ to learn about other command-line flags usable to configure the packaged app.
 To have high-resolution icons used by default for an app/domain, please
 contribute to the [icon repository](https://github.com/nativefier/nativefier-icons)!
 
+### Build Commands Library
```
Comment (ronjouch) [inline README.md:60]: Can you also add a concise mention about the library at the beginning of the README's `Troubleshooting` section?

**Review by ronjouch — CHANGES_REQUESTED** [review]
Thaaaaaaaaanks. A few small things:

**Comment by deciacco** [comment]
Thanks for your review...... I think I got it all as requested!

**Review by TheCleric — APPROVED** [review]

**Comment by deciacco** [comment]
Please let me know if I need to do something else to move forward...new to this... 😊👍

**Review by ronjouch — APPROVED** [review]
:clap: :clap: :clap: ! Merging.
